glean-core/ios/Glean/Glean.swift

```diff
@@ -105,7 +111,47 @@ public class Glean {
 
     /// Handle background event and send appropriate pings
     func handleBackgroundEvent() {
-        // sendPings()
+        // Perform task on a background thread
+        _ = Dispatchers.shared.launch {
```

Comment:
Why the dispatch here? `sendPingsByName` is already dispatched.
As I understand the `beginBackgroundTask` will make sure that the task stays around if the application is closed.

but the double-dispatch makes me uncomfortable. Would we do this for pretty much any `sendPing` triggered? e.g. also by those triggered on custom pings?

Reply:
You are right, it probably doesn't need this and could just be executed at this point.